#!/usr/bin/env python
PACKAGE = "qm_controllers"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_arm_wbc", double_t, 0, "Kp of arm wbc", 0.0, 0.0, 100)
gen.add("kd_arm_wbc", double_t, 0, "Kd of arm wbc", 0.5, 0.0, 100)

gen.add("kp_arm_1", double_t, 0, "Kp of arm joint 1", 1000, 0.0, 10000)
gen.add("kp_arm_2", double_t, 0, "Kp of arm joint 2", 6000, 0.0, 20000)
gen.add("kp_arm_3", double_t, 0, "Kp of arm joint 3", 5000, 0.0, 20000)
gen.add("kp_arm_4", double_t, 0, "Kp of arm joint 4", 500, 0.0, 10000)
gen.add("kp_arm_5", double_t, 0, "Kp of arm joint 5", 3000, 0.0, 10000)
gen.add("kp_arm_6", double_t, 0, "Kp of arm joint 6", 500, 0.0, 10000)

gen.add("kd_arm_1", double_t, 0, "Kd of arm joint 1", 8.0, 0.0, 100)
gen.add("kd_arm_2", double_t, 0, "Kd of arm joint 2", 30.0, 0.0, 100)
gen.add("kd_arm_3", double_t, 0, "Kd of arm joint 3", 8.0, 0.0, 100)
gen.add("kd_arm_4", double_t, 0, "Kd of arm joint 4", 0.2, 0.0, 100)
gen.add("kd_arm_5", double_t, 0, "Kd of arm joint 5", 0.2, 0.0, 100)
gen.add("kd_arm_6", double_t, 0, "Kd of arm joint 6", 0, 0.0, 100)

gen.add("arm_control_loop_hz", double_t, 0, "arm control loop hz", 200, 0.0, 1000) # 100

gen.add("d_arm_1", double_t, 0, "desired angle of arm joint 1", 3.1, -6.28, 6.28)
gen.add("d_arm_2", double_t, 0, "desired angle of arm joint 2", 3.7, -6.28, 6.28)
gen.add("d_arm_3", double_t, 0, "desired angle of arm joint 3", 0.9, -6.28, 6.28)
gen.add("d_arm_4", double_t, 0, "desired angle of arm joint 4", 2.7, -6.28, 6.28)
gen.add("d_arm_5", double_t, 0, "desired angle of arm joint 5", 1.4, -6.28, 6.28)
gen.add("d_arm_6", double_t, 0, "desired angle of arm joint 6", -0.4, -6.28, 6.28)

exit(gen.generate(PACKAGE, "weight", "Weight"))
